### PR TITLE
Cherry pick complex numbers support into main

### PR DESCRIPTION
**Pull request by annagrin**
### Description

Cherry-pick changes from https://github.com/NVIDIA/cuda-quantum/commit/a89e0d2f535a256603e3be0f420d9e89087fcc9f to main.

Note: I had to disable a couple of failing tests - not sure what is missing in main that causes them to fail.

### CODE REVIEW

File: python/tests/kernel/test_kernel_float.py [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+
+import sys
+
+import pytest
+import numpy as np
+
+import cudaq
+
+## [PYTHON_VERSION_FIX]
+skipIfPythonLessThan39 = pytest.mark.skipif(
```
Comment (khalatepradnya) [inline python/tests/kernel/test_kernel_float.py:17]: Ditto

**Review by khalatepradnya — COMMENTED** [review]
Q: Is the support for complex numbers already available in the builder mode?

**Comment by annagrin** [comment]
> Q: Is the support for complex numbers already available in the builder mode?

I think so - that is just using regular python.

**Comment by annagrin** [comment]
I am stashing this PR for now, got too hard with the failures in main to investigate. We have a workaround of using kernel builder for now.

**Comment by annagrin** [comment]
Reopened for failure investigation

**Review by schweitzpgi — COMMENTED** [review]
Skimming through this and it LGTM. Tests are still chugging though.

@khalatepradnya does this get you unblocked?

**Comment by khalatepradnya** [comment]
> Skimming through this and it LGTM. Tests are still chugging though.
> 
> @khalatepradnya does this get you unblocked?

Yes, it does. (I need one more commit pertaining to `np.exp()`)

**Review by schweitzpgi — APPROVED** [review]
Approving to unblock Pradnya.

**Pull request merged**